examples/make_bg_cube_models_true-reco_test.py

```diff
@@ -0,0 +1,217 @@
+
+
+def make_reco_model():
+    """Make a reco bg cube model."""
+
+    SCHEME = 'HESS'
+    METHOD = 'default'
+
+    fits_path = 'test_dataset'
+    overwrite = OVERWRITE
+    test = TEST
+    group_id = GROUP_ID
+
+    # create output folder
+    outdir = OUTDIR + '/reco'
+    if not os.path.isdir(outdir):
+        os.mkdir(outdir)
+    else:
+        if overwrite:
+            # delete and create again
+            shutil.rmtree(outdir) # recursively
+            os.mkdir(outdir)
+        else:
+            # do not overwrite, hence exit
+            s_error = "Cannot continue: directory \'{}\' exists.".format(outdir)
+            raise RuntimeError(s_error)
+
+    # 0. create dummy observation grouping
+    obs_groups = create_dummy_observation_grouping()
+    # save
+    outfile = outdir + '/bg_observation_groups.ecsv'
+    print('Writing {}'.format(outfile, overwrite=overwrite))
+    obs_groups.write(outfile)
+
+    # 1. create dummy dataset
+
+    observatory_name = SCHEME
+
+    # use enough stats so that rebinning (and resmoothing?) doesn't take place
+    n_obs = 100
+    if test:
+        # run fast
+        n_obs = 2
+
+    az_range = AZ_RANGE
+    alt_range = ALT_RANGE
+    random_state = np.random.RandomState(seed=0)
+
+    make_test_dataset(fits_path=fits_path, overwrite=overwrite,
+                      observatory_name='HESS', n_obs=n_obs,
+                      az_range=az_range,
+                      alt_range=alt_range,
+                      date_range=(Time('2010-01-01T00:00:00',
+                                       format='isot', scale='utc'),
+                                  Time('2015-01-01T00:00:00',
+                                       format='isot', scale='utc')),
+                      n_tels_range=(3, 4),
+                      random_state=random_state)
+
+    # 2. get observation table
+    scheme = SCHEME
+    data_store = DataStore(dir=fits_path, scheme=scheme)
+    observation_table = data_store.make_observation_table()
+    outfile = outdir + '/bg_observation_table_group{}.fits.gz'.format(group_id)
+    print("Writing {}".format(outfile))
+    observation_table.write(outfile, overwrite=overwrite)
+
+    # 3. build bg model
+    method = METHOD
+    bg_cube_model = make_bg_cube_model(observation_table=observation_table,
+                                       fits_path=fits_path,
+                                       method=method)
+
+    # save
+    outfile = outdir + '/bg_cube_model_group{}'.format(group_id)
+    print("Writing {}".format('{}_table.fits.gz'.format(outfile)))
+    print("Writing {}".format('{}_image.fits.gz'.format(outfile)))
+    bg_cube_model.write('{}_table.fits.gz'.format(outfile),
+                        format='table', clobber=overwrite)
+    bg_cube_model.write('{}_image.fits.gz'.format(outfile),
+                        format='image', clobber=overwrite)
+
+
+if __name__ == '__main__':
+
+    outdir = OUTDIR
+    overwrite = OVERWRITE
+
+    # create output folder
```

Comment:
These ~10 lines of code have appeared now a few times in your code I think.
Please make it a utility function.